contracts/EuroCashFlowLender.sol

```diff
@@ -0,0 +1,452 @@
+      quoteSignatureR,
+      quoteSignatureVS
+    );
+    (uint256 usdPayout, uint256 usdPremium, uint96 internalId) = _transformPolicyParams(
+      payout,
+      premium,
+      policyData
+    );
+
+    _increaseDebt(premium);
+    policyId = riskModule().newPolicy(
+      usdPayout,
+      usdPremium,
+      lossProb,
+      expiration,
+      address(this),
+      internalId
+    );
+    return policyId;
+  }
+
+  function _resolvePolicy(Policy.PolicyData calldata policy, uint256 payout) internal {
+    TrustfulRiskModule(address(policy.riskModule)).resolvePolicy(policy, payout);
+  }
+
+  function resolvePolicy(
+    Policy.PolicyData calldata policy,
+    uint256 payout
+  ) external onlyRole(RESOLVER_ROLE) {
+    payout = payout.wadMul(getEurUsdPrice());
+    _resolvePolicy(policy, payout);
+  }
+
+  function resolvePolicyFullPayout(
+    Policy.PolicyData calldata policy,
+    bool customerWon
+  ) external onlyRole(RESOLVER_ROLE) {
+    _resolvePolicy(policy, customerWon ? policy.payout : 0);
+  }
+
+  function onERC721Received(
+    address,
+    address,
+    uint256,
+    bytes calldata
+  ) external pure override returns (bytes4) {
+    return IERC721Receiver.onERC721Received.selector;
+  }
+
+  function onPolicyExpired(address, address, uint256) external pure override returns (bytes4) {
+    return IPolicyHolder.onPolicyExpired.selector;
+  }
+
+  /**
+   * @dev Whenever an Policy is resolved with payout > 0, this function is called
+   *
+   * It must return its Solidity selector to confirm the payout.
+   * If interface is not implemented by the recipient, it will be ignored and the payout will be successful.
+   * If any other value is returned or it reverts, the policy resolution / payout will be reverted.
+   *
+   * The selector can be obtained in Solidity with `IPolicyPool.onPayoutReceived.selector`.
+   *
+   */
+  function onPayoutReceived(
+    address,
+    address,
+    uint256,
+    uint256 amount
+  ) external override returns (bytes4) {
+    require(msg.sender == address(_pool()), "Only the PolicyPool should call this method");
+    uint256 payoutEUR = amount.wadDiv(getEurUsdPrice());
+    _decreaseDebt(payoutEUR);
+    return IPolicyHolder.onPayoutReceived.selector;
+  }
+
+  /**
+   * @dev Withdraws funds from the contract
+   *
+   * Can be executed by the owner to recover the funds.
+   *
+   * Requirements:
+   * - onlyRole(OWNER_ROLE)
+   *
+   * @param amount The amount to withdraw. If amount == type(uint256).max means all the funds
+   * @param destination The address that will receive the transferred funds.
+   * @return Returns the actual amount withdrawn.
+   */
+  function withdraw(
+    uint256 amount,
+    address destination
+  ) external onlyRole(OWNER_ROLE) returns (uint256) {
+    require(
+      destination != address(0),
+      "EuroCashFlowLender: destination cannot be the zero address"
+    );
+    require(_debt > 0, "EuroCashFlowLender: cannot withdraw when there is no debt");
```

Comment:
`require(_debt >= 0, "EuroCashFlowLender: cannot withdraw when there is no debt");`

Con deuda igual a cero sí podría retirar.